File: src/www/__test__/testSteps/alert/ThereIsAnAlertMessageThatSays.ts
import { AbstractPostLineStep } from "../AbstractPostLineStep";
import { getByAlert } from "./helpers";

export class ThereIsAnAlertMessageThatSays extends AbstractPostLineStep {
  constructor() {
    super(/There is an alert message that says "([^"]+)"/i);
  }

  async runMatch(_: any, match: string[]) {
    const [, message] = match;
    const alert = getByAlert(document.body);
    expect(alert).toHaveTextContent(message);
  }
}

File: src/test/java/com/drpicox/game/testSteps/alert/ThereIsAnAlertMessageThatSays.java
package com.drpicox.game.testSteps.alert;

import com.drpicox.game.testPost.reader.PostLine;
import com.drpicox.game.testSteps.AbstractPostLineStep;
import org.springframework.stereotype.Component;

@Component
public class ThereIsAnAlertMessageThatSays extends AbstractPostLineStep {

    @Override
    protected String getRegex() {
        return "There is an alert message that says \"([^\"]+)\"";
    }

    @Override
    protected void run(PostLine line, String[] match) {
        // No implementation required by tests in backend
    }
}
